### PR TITLE
add bootstrap method

### PR DESCRIPTION
**Pull request by robertjneal**

### CODE REVIEW

File: src/main/scala/com/udemy/statistics/algorithm/Sampling.scala [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import scala.util.Random
+import scalaz.NonEmptyList
+import scalaz.Scalaz._
+
+object Sampling {
+
+  def randomDraw[T:Numeric](x: NonEmptyList[T]): T =
+    x.index(Random.nextInt(x.length)).get
+
+  def drawSample[T: Numeric](x: Seq[T], size: Int): Seq[T] = {
```
Comment (pauljcho) [inline src/main/scala/com/udemy/statistics/algorithm/Sampling.scala]: how might a function that samples without replacement be related to this function? should we add a boolean argument like in R?

Comment (robertjneal) [reply src/main/scala/com/udemy/statistics/algorithm/Sampling.scala]: I considered that actually. I started adding the argument and then realized that I would then have to implement the sampling without replacement functionality and didn't want to do it in this PR. My thought is when we add that, we can make with replacement the default so as to maintain backward compatibility.

Comment (robertjneal) [reply src/main/scala/com/udemy/statistics/algorithm/Sampling.scala]: Okay, added support for sampling with replacement. Also, renamed per our offline conversation.

File: src/main/scala/com/udemy/statistics/algorithm/Sampling.scala [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  def sample[T: Numeric](x: Seq[T], size: Int, withReplacement: Boolean = true): Seq[T] = {
+    x.toList.toNel match {
+      case None => Seq.empty
+      case Some(nel) =>
+        if (withReplacement) ParVector.fill(size)(randomDraw(nel)).toVector
+        else Random.shuffle(x).take(size)
+    }
+  }
+
+  def bootstrap[T: Numeric](sample: Seq[T], draws: Int, statistic: Seq[T] => Double): Seq[Double] =
```
Comment (pauljcho) [inline src/main/scala/com/udemy/statistics/algorithm/Sampling.scala]: nit: what do you think about being consistent with the arg name for the data being passed into these functions? here's it's `sample`, but for the others it's `x`

Comment (robertjneal) [reply src/main/scala/com/udemy/statistics/algorithm/Sampling.scala]: I'm not sure, but I think my thinking was that you always bootstrap a sample, but you can use the other methods for any collection. For example, it makes sense to randomly draw from a population, but it doesn't make sense to bootstrap a population.

Comment (pauljcho) [reply src/main/scala/com/udemy/statistics/algorithm/Sampling.scala]: i see. makes sense

File: src/main/scala/com/udemy/statistics/algorithm/Sampling.scala [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+import scala.Numeric.Implicits._
+import scala.collection.parallel.immutable.ParVector
+import scala.util.Random
+import scalaz.NonEmptyList
+import scalaz.Scalaz._
+
+object Sampling {
+
+  def randomDraw[T:Numeric](x: NonEmptyList[T]): T =
```
Comment (pauljcho) [inline src/main/scala/com/udemy/statistics/algorithm/Sampling.scala]: <sup>nit</sup>: `T: Numeric` (with the space) for consistency?

Comment (robertjneal) [reply src/main/scala/com/udemy/statistics/algorithm/Sampling.scala]: Ah, thanks. Not sure how these ended up inconsistent.

Comment (robertjneal) [reply src/main/scala/com/udemy/statistics/algorithm/Sampling.scala]: fixed

**Pull request merged**